src/browser/tab/mod.rs

```diff
@@ -43,6 +43,18 @@ pub type RequestInterceptor = Box<
         + Sync,
 >;
 
+#[rustfmt::skip]
+pub type ResponseHandler = Box<
```

Comment:
Shouldn't this be a trait?

Reply:
Yeah, it should.

I was just sticking to how the intercept handler is defined.

Happy to introduce a trait instead.

The bigger question for me is: the tests are a bit flaky in general, and especially my new test. Sometimes passes, sometimes fails. Something fails in the wait for navigation handling. 

I won't have time for a few days to dig into this so any advice is appreciated.

Reply:
Running tests on master over and over now, hopefully I can trigger a failure.

Reply:
Here's my little notebook I'm keeping (mainly for myself) as I work through this: https://paper.dropbox.com/doc/Navigation-in-headless_chrome--AgCNa0JHNG4ouDv5R_m~2Ip6AQ-viXQEuMOi3HXHeD58oZYi

Reply:
@lukaslue do you think this should block a merge and release? and do you think `Tab.request_interceptor` should be a trait too? I'm not sure of the best way to do it, otherwise I would now.

Reply:
https://github.com/atroche/rust-headless-chrome/issues/137